impl/src/main/java/com/okta/sdk/impl/resource/AbstractPropertyRetriever.java

```diff
@@ -279,7 +279,15 @@ protected Map getMapProperty(String key) {
 
         if (value != null) {
             if (value instanceof String) {
-                return enumConverter.fromValue(type, value.toString());
+                E convertedValue;
+                try {
+                    convertedValue = enumConverter.fromValue(type, value.toString());
+                } catch (IllegalArgumentException e) {
+                    convertedValue = enumConverter.fromValue(type, "SDK_UNKNOWN");
+                    String msg = "Undeclared enum value. Defaulting to SDK_UNKNOWN";
+                    log.error(msg, e);
```

Comment:
This isn't an "error" as it is recoverable, it should probably be a "warn", the error message should include the original string value as well, and probably some indication of how the user should access the value i.e. `.get("key", String.class)` (where `key` is the actual key string if possible